### PR TITLE
Fix `.strip()` to strip all whitespace characters

### LINKED ISSUE #4188: Unicode lstrip/rstrip/strip do not handle all whitespace characters
## Reporting a bug

- [x] I am using the latest ~released~ master version of Numba (most recent is visible in
 the change log (https://github.com/numba/numba/blob/master/CHANGE_LOG).
- [x] I have included below a minimal working reproducer (if you are unsure how
 to write one see http://matthewrocklin.com/blog/work/2018/02/28/minimal-bug-reports).

Unicode `lstrip`, `rstrip`, `strip` functions only handle space as whitespace. Other whitespace characters like `\t` are ignored:
```python
In [5]: numba.njit(lambda: '\ta'.lstrip())()                                                                                                                                                                
Out[5]: '\ta'

In [6]: '\ta'.lstrip()                                                                                                                                                                                      
Out[6]: 'a'
```
`string` library reports whitespace characters as below, but we need to look into CPython implementation.
```
In [1]: import string                                                                                                                                                                                       
In [2]: string.whitespace                                                                                                                                                                                   
Out[2]: ' \t\n\r\x0b\x0c'
```

### LINKED ISSUE #4731: `.strip()` not stripping `\t`.
## Reporting a bug

- [x] I am using the latest released version of Numba (most recent is visible in
 the change log (https://github.com/numba/numba/blob/master/CHANGE_LOG).
- [x] I have included below a minimal working reproducer (if you are unsure how
 to write one see http://matthewrocklin.com/blog/work/2018/02/28/minimal-bug-reports).

```
>>> from numba import njit
>>> s = '  \t bbcd\t '
>>> s.strip()
'bbcd'
>>> @njit
... def foo(s):
...     return s.strip()
...
>>> foo(s)
'\t bbcd\t'
>>>

```

### PR DESCRIPTION
**Pull request by rohitsanj**
fixes #4188
(also fixes #4731)

Changes:
- Replaced `' '` with `' \t\n\r\x0b\x0c'` to check for all whitespace characters
- I have added tests as well

To see it working -
```python
from numba import njit
@njit 
def foo(s):
   return s.strip()

print(foo('\t abcd \t'))
```

### CODE REVIEW

**Comment by esc** [comment]
@rohitsanj thanks for opening the PR and congrats on submitting your first improvement to Numba! I have queued the PR for review now.

**Comment by stuartarchibald** [comment]
Thanks for the PR. I think this probably ought to use `numba.core.cpython.unicode_support::_PyUnicode_IsSpace()` for the unicode path and `numba.core.cpython.unicode_support::_Py_ISSPACE()` on the ASCII fast-path?

**Comment by rohitsanj** [comment]
> Thanks for the PR. I think this probably ought to use `numba.core.cpython.unicode_support::_PyUnicode_IsSpace()` for the unicode path and `numba.core.cpython.unicode_support::_Py_ISSPACE()` on the ASCII fast-path?

I'm not exactly sure how and where I would use these functions. Can you please give me some more info?

**Comment by shimst3r** [comment]
@rohitsanj thanks a lot for taking care of this issue. I totally forgot about it. 🙇

**Comment by stuartarchibald** [comment]
Answered here https://gitter.im/numba/numba?at=5e3d9609df1153705e1f82bb

**Comment by rohitsanj** [comment]
@stuartarchibald As discussed over Gitter, are we going to merge this now and then later come up with the optimized implementation?

**Comment by stuartarchibald** [comment]
@rohitsanj I think this still needs to use `_PyUnicode_IsSpace`, the "optimisation" of adding an ASCII fast path can be done later. The current PR does not encompass all whitespace as per the whitespace supported by CPython's unicode implementation. For example, this fails:
```python
from numba import njit
import numpy as np

@njit
def foo():
    x = '\u2029' # Unicode Character 'PARAGRAPH SEPARATOR'
    return len(x.strip())

assert foo() == foo.py_func()
```
switching to use the noted function will fix all this.

**Comment by rohitsanj** [comment]
Got it. Will update this PR soon.

**Comment by rohitsanj** [comment]
@stuartarchibald Please review it now. Let me know if I need to add more test cases 😄

File: numba/tests/test_unicode.py [hunk 1]
```diff
@@ -1701,7 +1701,12 @@ def test_strip(self):
             ('  tú quién te crees?   ', None),
             ('大处 着眼，小处着手。大大大处', '大处'),
             (' 大处大处  ', ''),
-            (' 大处大处  ', None)
+            (' 大处大处  ', None),
+            ('\t abcd \t', None),
+            ('\n abcd \n', None),
+            ('\r abcd \r', None),
+            ('\x0b abcd \x0b', None),
+            ('\x0c abcd \x0c', None)
```
Comment (stuartarchibald) [inline numba/tests/test_unicode.py]: Perhaps also add a version of example here https://github.com/numba/numba/pull/5213#issuecomment-584062365 with a unicode kind 4 whitespace either side of some ASCII? e.g. `'\u2029abcd\u205F'` ?

Comment (stuartarchibald) [reply numba/tests/test_unicode.py]: And also add in some variants where the `chars` kwarg is something other than `None`?

Comment (rohitsanj) [reply numba/tests/test_unicode.py]: I have added two more cases, please take a look.

Comment (rohitsanj) [reply numba/tests/test_unicode.py]: Done. Added a case where `chars` is something other than `None`

**Comment by stuartarchibald** [comment]
@rohitsanj Congratulations on your first contribution to Numba!

**Comment by rohitsanj** [comment]
Thanks for all your help @stuartarchibald 😄

**Pull request merged**